docs/data/date-pickers/base-concepts/ServerInteraction.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import Divider from '@mui/material/Divider';
+import Button from '@mui/material/Button';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateField, DateFieldProps } from '@mui/x-date-pickers/DateField';
+import useControlled from '@mui/utils/useControlled';
+
+type DisplayEventsProps = {
+  logsFromOnChange: (Dayjs | null)[];
+  logsFromOnAccept: (Dayjs | null)[];
+};
+function DisplayEvents(props: DisplayEventsProps) {
+  const { logsFromOnChange, logsFromOnAccept } = props;
+  return (
+    <Stack direction="row" justifyContent="space-between" sx={{ p: 2 }}>
+      <Stack sx={{ minWidth: 180 }}>
+        <Typography>onChange</Typography>
+        <Divider />
+        <Typography component="pre" sx={{ maxHeight: '15rem', overflow: 'auto' }}>
```

Comment:
Have you considered extracting this layout into another component as well?
IMHO it would make sense and at the same time—it would avoid the issue of `onAccept` section not having maxHeight and increasing the container height after enough delayed field modifications 🙈 
![Screenshot 2023-04-17 at 13 16 06](https://user-images.githubusercontent.com/4941090/232456069-7adc3e8f-83e8-44d0-9308-60eebbebf5ef.png)